### PR TITLE
Added a "static" IPSource

### PR DESCRIPTION
**Pull request by otaconix**
I personally needed a way for caddy-dynamicdns to create records for IP addresses it could not otherwise acquire from other sources (it's running in Docker, with an IPv4-only bridge, and the host has static routable IPv6 addresses).

This PR is mostly there to gauge if @mholt would even consider merging this, as I understand it may broaden the scope of caddy-dynamicdns a bit. If so, there's docs to be updated, and undoubtedly code improvements to be had, so let me know!

### CODE REVIEW

File: ipsource.go [hunk 1]
```diff
@@ -307,6 +308,74 @@ func (u NetInterface) GetIPs(ctx context.Context, versions IPVersions) ([]net.IP
...
+	d.Next() // skip directive name
+
+	for d.NextArg() {
+		raw_ip := d.Val()
+		ip := net.ParseIP(raw_ip)
+
+		if ip == nil {
+			s.invalid_ips = append(s.invalid_ips, raw_ip)
+			continue
+		}
```
Comment (francislavoie) [inline ipsource.go:371]: This is weird IMO, just return an error if any invalid IPs are configured.

**Comment by francislavoie** [comment]
I think this is fine to add, it's pretty simple and doesn't really cause any trouble. I just think the warning thing is weird, I think it should just be an error if invalid IPs are passed.

File: ipsource.go (hunk 1 above)
Comment (otaconix) [reply ipsource.go]: Is this better?

**Review by mholt — APPROVED** [review]
Sure, this LGTM now. Thanks!

File: ipsource.go (hunk 1 above)
Comment (mholt) [reply ipsource.go]: I think so!

**Pull request merged**